packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -27,16 +28,31 @@ function SelectionTool(props: Props) {
     id = 'Selection',
     modifierKey,
     disabled,
+    clampCenter,
   } = props;
 
   const camera = useThree((state) => state.camera);
   const { worldToData } = useAxisSystemContext();
   const shouldInteract = useRegisterInteraction(id, { modifierKey, disabled });
 
-  const [startPoint, setStartPoint] = useState<Vector2>();
-  const [endPoint, setEndPoint] = useRafState<Vector2 | undefined>(undefined);
+  const [originPoint, setOriginPoint] = useState<Vector2>();
```

Comment:
As the `originPoint` (that is the point where the user clicked) can be different from the `startPoint`, it must be stored separately. I also store it in world coordinates for reasons exposed below.